client/asset/eth/eth.go

```diff
@@ -3691,6 +3749,59 @@ func (w *assetWallet) checkFindRedemptions() {
 	}
 }
 
+// sumPendingTxs sums the expected incoming and outgoing values in unconfirmed
+// transactions stored in pendingTxs. Not used if the node is a mempoolFetcher.
+func (w *assetWallet) sumPendingTxs() (out, in uint64) {
+	w.tipMtx.RLock()
+	tip := w.currentTip.Number.Uint64()
+	w.tipMtx.RUnlock()
+
+	isETH := w.assetID == BipID
+	addPendingSend := func(pt *pendingTx) {
+		in += pt.in
+		if isETH {
+			if pt.assetID == BipID {
+				out += pt.out + pt.fees
+			} else {
+				out += pt.fees
+			}
+			return
+		}
+		// token
+		out += pt.out
+	}
+
+	w.pendingTxMtx.Lock()
+	defer w.pendingTxMtx.Unlock()
+	for txHash, pt := range w.pendingTxs {
+		if !isETH && pt.assetID != w.assetID {
+			continue
+		}
+		if pt.lastCheck == tip {
+			// Expect nothing has changed since our last check.
+			addPendingSend(pt)
+			continue
+		}
+		confs, err := w.node.transactionConfirmations(w.ctx, txHash)
+		if err != nil && !errors.Is(err, asset.CoinNotFoundError) {
+			w.log.Errorf("Error getting confirmations for pending tx %s: %v", txHash, err)
+			// DRAFT NOTE: Should I stop tracking this tx? I'd hate to end up in
+			// a state where we're spamming logs every block for a tx that will
+			// never happen, though presumably that would result in
+			// CoinNotFoundError.
+			// delete(w.pendingTxs, txHash)
```

Comment:
On this note, I think we need to delete the old tx in `resubmitRedemption` (or `recordReplacementTx` if we modify that method to work with a nil `*monitoredTx`).

But I think I see ways, mostly due to racing with the blockchain, how a pending tx in this map could disappear, so I also feel like we might end up needing a stamp on each `pendingTx`.  We might want to try without it first before we get more complex.

Reply:
@martonp you know this replacement code best.  Pls see if there's a need to handle things there.

Reply:
Yeah if `resubmitRedemption` has been called then we have determined that the tx will definitely no longer be mined, so it can be removed from the pending txs list. Then the call to `Redeem` will add the replacement to the list.

However, if in this function we check if the nonce of a `pendingTx` has been used by another transaction and remove it from the `pendingTxs` list if so, then there's no need to update `resubmitRedemption`.